covid19risk/permission-portal
# Make first time users update password

Closes #214 

Prompt the user to set a new password when they log in if the `isFirstTimeUser` flag is set.

To test:
1. Make sure that the cloud functions from this PR are on the server.
2. Create a new user. It will have `isFirstTimeUser: true`.
3. Log in as the new user from the home page.
4. In the `ChangePasswordModal`, enter a new password and click "Save".

Notes:
- Firebase requires passwords to be 6 characters long and throws `auth/weak-password` if the submitted password is not strong enough. We enforce this requirement on the client side by disabling the submit button unless it is met.
- Firebase throws `auth/requires-recent-login` if the user logged in too long before a call to [`updatePassword()`][updatePassword]. We handle this by prompting the user to [reauthenticate] using their current password and retry the operation.

[updatePassword]: https://firebase.google.com/docs/reference/js/firebase.User#updatepassword
[reauthenticate]: https://firebase.google.com/docs/reference/js/firebase.User#reauthenticatewithcredential
